### PR TITLE
Optimizations and Fixes

### PR DESCRIPTION
**Pull request by thevindu-w**

### CODE REVIEW

File: src/server/JasmineGraphInstanceService.cpp [hunk 1]
```diff
@@ -931,7 +939,7 @@ void JasmineGraphInstanceService::trainPartition(string trainData) {
 
     std::string path = "cd " + Utils::getJasmineGraphProperty("org.jasminegraph.graphsage") + " && ";
     std::string command =
-        path + "python3.8 -m unsupervised_train >  /home/ubuntu/software/jasminegraph/logs/unsupervised_train" +
+        path + "python3 -m unsupervised_train >  /home/ubuntu/software/jasminegraph/logs/unsupervised_train" +
```
Comment (miyurud) [inline src/server/JasmineGraphInstanceService.cpp:942]: Lets change the path where the log file getting created from `/home/ubuntu/software/jasminegraph/logs` to `/var/tmp/jasminegraph/logs`. Latter is the standard location we maintain runtime logs.

File: src/server/JasmineGraphInstanceService.cpp [hunk 2]
```diff
@@ -1804,29 +1831,30 @@ map<long, unordered_set<long>> getEdgesWorldToLocal(string graphID, string parti
...
+        map<long, unordered_set<long>> centralGraphMap = centralStore.getUnderlyingHashMap();
+
+        for (map<long, unordered_set<long>>::iterator centralGraphMapIterator = centralGraphMap.begin();
+             centralGraphMapIterator != centralGraphMap.end(); ++centralGraphMapIterator) {
+            long startVid = centralGraphMapIterator->first;
+            unordered_set<long> endVidSet = centralGraphMapIterator->second;
+
+            for (auto itr = endVidSet.begin(); itr != endVidSet.end(); ++itr) {
+                if (graphVertexMap.find(*itr) != graphVertexMap.end()) {
+                    map<long, unordered_set<long>>::iterator toIDiterator = worldToLocalVertexMap.find(*itr);
```
Comment (miyurud) [inline src/server/JasmineGraphInstanceService.cpp:1848]:
```suggestion
                    map<long, unordered_set<long>>::iterator toIDIterator = worldToLocalVertexMap.find(*itr);
```

File: src/server/JasmineGraphInstanceService.cpp [hunk 3]
```diff
@@ -2151,541 +2181,160 @@ static void batch_upload_command(int connFd, bool *loop_exit_p) {
...
+
+    std::vector<string> workerSockets;
+    stringstream wl(workerList);
+    string intermediate;
+    while (getline(wl, intermediate, ',')) {
+        workerSockets.push_back(intermediate);
+    }
+
+    JasmineGraphInstanceService::duplicateCentralStore(serverPort, stoi(graphID), stoi(partitionID), workerSockets,
+                                                       "localhost");
```
Comment (miyurud) [inline src/server/JasmineGraphInstanceService.cpp:2293]: We are sending `localhost` here. Is this acceptable? Can't there be scenarios where we need to use other host other than `localhost`?

**Review by miyurud — CHANGES_REQUESTED** [review]
I have put some comments. Please address them all.

File: src/server/JasmineGraphInstanceService.cpp (hunk 2 above)
Comment (thevindu-w) [reply src/server/JasmineGraphInstanceService.cpp:1848]: Done

File: src/server/JasmineGraphInstanceService.cpp (hunk 3 above)
Comment (thevindu-w) [reply src/server/JasmineGraphInstanceService.cpp]: Done

File: src/server/JasmineGraphInstanceService.cpp (hunk 1 above)
Comment (thevindu-w) [reply src/server/JasmineGraphInstanceService.cpp:956]: Done

File: src/server/JasmineGraphInstanceService.cpp (hunk 2 above)
Comment (miyurud) [reply src/server/JasmineGraphInstanceService.cpp]: Ok

File: src/server/JasmineGraphInstanceService.cpp (hunk 3 above)
Comment (miyurud) [reply src/server/JasmineGraphInstanceService.cpp]: Ok

**Review by miyurud — APPROVED** [review]

**Pull request merged**